extension/js/common/xss-safe-factory.ts

```diff
@@ -233,8 +233,12 @@ export class XssSafeFactory {
       const btn = `<div class="new_secure_compose_window_button" id="flowcrypt_secure_compose_button" title="New Secure Email"><img src="${this.srcImg('logo-19-19.png')}"></div>`;
       return `<div class="_fce_c ${this.destroyableCls} cryptup_compose_button_container" role="presentation">${btn}</div>`;
     } else {
-      const btn = `<div class="new_secure_compose_window_button" id="flowcrypt_secure_compose_button" role="button" tabindex="0" data-test="action-secure-compose">Secure Compose</div>`;
-      return `<div class="${this.destroyableCls} z0">${btn}</div>`;
+      let btn = `<div class="new_secure_compose_window_button" id="flowcrypt_secure_compose_button" role="button" tabindex="0" data-test="action-secure-compose">Secure Compose</div>`;
+      if ($('.V6.CL.W9').length === 1) { // if mail primary button present for new Gmail UI
+        btn = `<div class="new_secure_compose_window_button compose_button_simple" id="flowcrypt_secure_compose_button" role="button" tabindex="0" data-test="action-secure-compose">
+        </div><div class="apW">FlowCrypt</div>`;
+      }
+      return `<div class="${this.destroyableCls} z0 ${$('.V6.CL.W9').length === 1 ? 'pb-10px' : '' }">${btn}</div>`;
```

Comment:
I'd suggest to create a `const isNewGmailUI = $('.V6.CL.W9').length === 1;`
and use it in both places
so we don't end up fixing the first case and forgetting the other one if something changes.

Reply:
Agree, let's call it `isNewGmailUi2022`

Reply:
Thank you, sir, @rrrooommmaaa, and sir @tomholub for the review. 

I've also added a few css tweaks to the UI to maintain it's signature "secure compose" look while adapting to the new ui:
<img width="68" alt="Screen Shot 2022-03-08 at 9 00 57 AM" src="https://user-images.githubusercontent.com/46025304/157145349-5d2a78e8-dab8-4c8e-a569-63ab520b79d5.png">